internal/wasm/sdk/internal/wasm/vm.go

```diff
@@ -581,3 +626,23 @@ func callVoid(vm *VM, name string, args ...int32) error {
 	_, err := vm.instance.GetExport(name).Func().Call(sl...)
 	return err
 }
+
+func (i *VM) clearInterrupts(ctx context.Context) error {
```

Comment:
For re-review, please just focus on this: the "clear interrupts" logic added to avoid falling over spurious traps set from previous runs.

Reply:
This seems like a necessary evil. The simplest approach for us would be to only call the interrupt handler if the VM was executing. Unfortunately, the cancellation goroutine has no way of knowing if the VM is executing. This allows the cancellation goroutine to race and call the interrupt handler when it's not necessary.